File: cpp/1-bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

void print(vector<int>& v);

void bubbleSort(vector<int>& v) {
	const int n = (int) v.size();

	for(int i = 0; i < n - 1; ++i) {
		bool swapped = false;
		for(int j = 0; j < n - i - 1; ++j) {
			if(v[j] > v[j + 1]) {
				swap(v[j], v[j + 1]);
				swapped = true;
			}
		}

		if(!swapped)
			break;
	}
}

int main() {
	vector<int> t {3, 1, 5, 2, 4};
	print(t); 
	bubbleSort(t);
	print(t);

	return 0;
}

void print(vector<int>& v) {
	for(auto& el: v) cout << el << ' ';
	cout << '\n';
}

File: java/1-bubbleSort.java
import java.util.*;

public class Program {

	static void print(int v[]) {
		final int N = v.length;
		for(int el: v) {
			System.out.print(el + " ");
		}
		System.out.println();
	}

	static void bubbleSort(int v[]) {
		final int N = v.length;

		for(int i = 0; i < n - 1; ++i) {

			boolean swapped = false;
			for(int j = 0; j < n - i - 1; ++j) {
				if(v[j] > v[j + 1]) { 
					// swapping
					int t = v[j];
					v[j] = v[j + 1];
					v[j + 1] = t;
					swapped = true;
				}
			}

			if(!swapped) 
				break;
		}

	}

	public static void main(String[] args) {
		int[] v = {3, 1, 5, 2, 4};
		print(v);
		bubbleSort(v);
		print(v);
	}
}